app/src/main/java/com/keylesspalace/tusky/components/timeline/TimelineFragment.kt

```diff
@@ -0,0 +1,562 @@
+        viewModel.changeContentCollapsed(isCollapsed, position)
+    }
+
+    override fun onViewMedia(position: Int, attachmentIndex: Int, view: View?) {
+        val status = viewModel.statuses[position].asStatusOrNull() ?: return
+        super.viewMedia(
+            attachmentIndex,
+            AttachmentViewData.list(status.actionable),
+            view
+        )
+    }
+
+    override fun onViewThread(position: Int) {
+        val status = viewModel.statuses[position].asStatusOrNull() ?: return
+        super.viewThread(status.actionable.id, status.actionable.url)
+    }
+
+    override fun onViewTag(tag: String) {
+        if (viewModel.kind == TimelineViewModel.Kind.TAG && viewModel.tags.size == 1 &&
+            viewModel.tags.contains(tag)
+        ) {
+            // If already viewing a tag page, then ignore any request to view that tag again.
+            return
+        }
+        super.viewTag(tag)
+    }
+
+    override fun onViewAccount(id: String) {
+        if ((viewModel.kind == TimelineViewModel.Kind.USER ||
+                    viewModel.kind == TimelineViewModel.Kind.USER_WITH_REPLIES) &&
+            viewModel.id == id
+        ) {
+            /* If already viewing an account page, then any requests to view that account page
+             * should be ignored. */
+            return
+        }
+        super.viewAccount(id)
+    }
+
+    private fun onPreferenceChanged(key: String) {
+        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+        when (key) {
+            PrefKeys.FAB_HIDE -> {
+                hideFab = sharedPreferences.getBoolean(PrefKeys.FAB_HIDE, false)
+            }
+            PrefKeys.MEDIA_PREVIEW_ENABLED -> {
+                val enabled = accountManager.activeAccount!!.mediaPreviewEnabled
+                val oldMediaPreviewEnabled = adapter.mediaPreviewEnabled
+                if (enabled != oldMediaPreviewEnabled) {
+                    adapter.mediaPreviewEnabled = enabled
+                    updateViews()
+                }
+            }
+        }
+    }
+
+    public override fun removeItem(position: Int) {
+        viewModel.statuses.removeAt(position)
+        updateViews()
+    }
+
+    private fun onLoadMore() {
+        viewModel.loadMore()
+    }
+
+    private fun actionButtonPresent(): Boolean {
+        return viewModel.kind != TimelineViewModel.Kind.TAG &&
+                viewModel.kind != TimelineViewModel.Kind.FAVOURITES &&
+                viewModel.kind != TimelineViewModel.Kind.BOOKMARKS &&
+                activity is ActionButtonActivity
+    }
+
+    private fun updateViews() {
+        differ.submitList(viewModel.statuses.toList())
+
+        if (isAdded) {
+            binding.swipeRefreshLayout.isRefreshing = viewModel.isRefreshing
+            binding.progressBar.visible(viewModel.isLoadingInitially)
+            // TODO: ??
+            binding.swipeRefreshLayout.isEnabled = true
```

Comment:
I still don't know what to do with this one

Reply:
what is the problem?

Reply:
I have no idea when we should enable/disable it

Reply:
I think it should be enabled as long as no other way of refreshing is available. So it should only be disabled while the fullscreen statusview with a retry button is shown. Having both options is weird imho.